source/game/system/map/MapdataCannonPoint.cc

```diff
@@ -0,0 +1,45 @@
+#include "MapdataCannonPoint.hh"
+
+#include "game/system/CourseMap.hh"
+
+namespace System {
+
+MapdataCannonPoint::MapdataCannonPoint(const SData *data) : m_rawData(data) {
+    u8 *unsafeData = reinterpret_cast<u8 *>(const_cast<SData *>(data));
+    EGG::RamStream stream = EGG::RamStream(unsafeData, sizeof(SData));
+    read(stream);
+}
+
+void MapdataCannonPoint::read(EGG::Stream &stream) {
+    m_pos.read(stream);
+    m_rot.read(stream);
+    m_id = stream.read_u16();
+    m_parameterIdx = stream.read_s16();
+}
+
+const EGG::Vector3f &MapdataCannonPoint::pos() const {
+    return m_pos;
+}
+
+const EGG::Vector3f &MapdataCannonPoint::rot() const {
+    return m_rot;
+}
+
+u16 MapdataCannonPoint::id() const {
+    return m_id;
+}
+
+s16 MapdataCannonPoint::parameterIdx() const {
+    return m_parameterIdx;
+}
+
+/// @addr {Inlined at 0x80512FA4}
```

Comment:
Ugh sorry, I forgot to include this change when I actually made this comment before. This space prevents the address graphic from being generated.
```suggestion
/// @addr{Inlined at 0x80512FA4}
```